handbook.md

```diff
@@ -0,0 +1,233 @@
+---
+layout: contribute
+title: Handbook
+---
+
+# How we make things
+
+## Branching and SCM
+
+We use git for source control. Our projects in
+[theforeman](https://github.com/theforeman) follow this model for branching:
+
+* **develop**
+    * All pull requests are merged here.
+    * Contains nightlies code. It's relatively stable, but we make no promises about it.
+    * [`test_develop` in Jenkins](http://ci.theforeman.org/job/test_develop/) ensures our tests are passing at all times for various configurations.
+* **release-stable (i.e. 1.10-stable)** -
+    * Code is cherry-picked using `git cherry-pick -m` into this branch.
+    * It contains the latest stable code for a release cycle, 1.8-stable contains 1.8.1, 1.8.2, 1.8.3...
+    * Typically, we will announce the start of a new branch
+
+And we tag releases:
+
+* **release (i.e. 1.10)**
+    * Code is cherry-picked using `git cherry-pick -m` into this tag.
+    * It contains the latest stable code for a *specific release*, for instance, 1.10.0.
+
+We do not force push to any of these branches. If a commit must be reverted, use `git-revert`, and push the reverted commit.
+
+Foreman projects strive to have a linear history ([like this](http://i.imgur.com/RMFS3bt.png)). A way to achieve this is to never create merge commits, by always rebasing your changes with develop `git pull --rebase upstream develop`. This allows us to easily revert our code to any point in time, which is helpful to detect in what commit we introduced an error using `git-bisect`.
+
+## Changelog
+
+TODO - Try to describe how this looks https://www.mozilla.org/en-US/firefox/39.0/releasenotes/
+
+## Coding standards
+
+The purpose of the Foreman coding standards is to create a baseline for collaboration and review within various aspects of the Foreman project and community, from core code to plugins.
+
+Coding standards help avoid common coding errors, improve the readability of code, and simplify modification. They ensure that files within the project appear as if they were created by a single person.
+
+### Pull requests
+
+All pull requests need to have an associated issue in the Foreman [issue tracker](http://projects.theforeman.org/).
+
+Foreman's [PR processor](https://github.com/theforeman/prprocessor) will parse all pull requests, assign labels, and run tests for all major projects.
+Pull requests are always *rebased* on top of the develop branch so that the git log stays linear.
+
+### Commit messages
+
+Provide a brief description of the change in the first line (50 chars or less), including a issue number.
+
+The title must start with ```Fixes #xxxx```. If your fix essentially extends a previous fix, use ```Refs #xxxx```. Our PR processor will understand these formats and will auto associate the pull request with the issue in the tracker.
```

Comment:
I edited http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches-commit_message_format recently to clarify use of "refs" a little, i.e. it should only be used to add to tickets that haven't shipped yet.  I don't know if you want to mention that here?  The motivation is that we can't track new commits being added to closed tickets against closed releases in Redmine.